modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/reagents/succubus_milk.dm

```diff
@@ -99,93 +99,140 @@
+				
+			var/obj/item/organ/external/genital/vagina/new_vagina = new
+			new_vagina.build_from_dna(exposed_mob.dna, ORGAN_SLOT_VAGINA)
+			new_vagina.Insert(exposed_mob, 0, FALSE)
+			exposed_mob.update_body()
+			if(!incubus_draft)
+				to_chat(exposed_mob, span_purple("You feel a warmth in your groin as something blossoms down there."))
+				
+		if(!exposed_mob.getorganslot(ORGAN_SLOT_WOMB))
+			if (exposed_mob.dna.mutant_bodyparts[ORGAN_SLOT_WOMB][MUTANT_INDEX_NAME] == "None")
+				exposed_mob.dna.mutant_bodyparts[ORGAN_SLOT_WOMB][MUTANT_INDEX_NAME] = "Normal"
+				
+			var/obj/item/organ/external/genital/womb/new_womb = new
+			new_womb.build_from_dna(exposed_mob.dna, ORGAN_SLOT_WOMB)
+			new_womb.Insert(exposed_mob, 0, FALSE)
+			exposed_mob.update_body()
+	
+	// Separates gender change stuff from breast growth and shrinkage, as well as from new genitalia growth/removal
 	if(exposed_mob.client?.prefs.read_preference(/datum/preference/toggle/erp/gender_change))
-		var/obj/item/organ/external/genital/penis/mob_penis = exposed_mob.getorganslot(ORGAN_SLOT_PENIS)
-		var/obj/item/organ/external/genital/testicles/mob_testicles = exposed_mob.getorganslot(ORGAN_SLOT_TESTICLES)
-		if(exposed_mob.gender == MALE)
+		if (incubus_draft)
+			if(exposed_mob.gender != PLURAL)
+				exposed_mob.set_gender(PLURAL)
+				exposed_mob.physique = exposed_mob.gender
+				exposed_mob.update_body()
+				exposed_mob.update_mutations_overlay()
+		else if(exposed_mob.gender != FEMALE)
 			exposed_mob.set_gender(FEMALE)
 			exposed_mob.physique = exposed_mob.gender
 			exposed_mob.update_body()
 			exposed_mob.update_mutations_overlay()
-		if(!mob_penis)
-			return
-		if(exposed_mob.client?.prefs.read_preference(/datum/preference/toggle/erp/penis_enlargement)) // To do cock shrinkage, check if prefs allow for this.
-			if(mob_penis.genital_size > penis_min_length)
-				mob_penis.genital_size -= penis_size_reduction_step
-				mob_penis.update_sprite_suffix()
-				exposed_mob.update_body()
-			if(mob_penis.girth > penis_minimum_girth)
-				mob_penis.girth -= penis_girth_reduction_step
-				mob_penis.update_sprite_suffix()
+		
+	// Cock & ball shrinkage portion. Check if prefs allow for this.
+	var/obj/item/organ/external/genital/penis/mob_penis = exposed_mob.getorganslot(ORGAN_SLOT_PENIS)
+	var/obj/item/organ/external/genital/testicles/mob_testicles = exposed_mob.getorganslot(ORGAN_SLOT_TESTICLES)
+	if(exposed_mob.client?.prefs.read_preference(/datum/preference/toggle/erp/penis_shrinkage))
+		// Penis shrinkage
+		if(mob_penis)
+			// Handle completely shrinking away, if prefs allow
+			if(mob_penis.genital_size == penis_min_length)
+				if(exposed_mob.client?.prefs?.read_preference(/datum/preference/toggle/erp/genitalia_removal))
+					mob_penis.Remove(exposed_mob)
+					exposed_mob.update_body()
+			else 
+				if(mob_penis.genital_size > penis_min_length)
+					mob_penis.genital_size -= penis_size_reduction_step
```

Comment:
Do this instead, that will ensure that the size is properly capped. You should do this every where there's size changes.
```suggestion
					mob_penis.genital_size = max(mob_penis.genital_size - penis_size_reduction_step, penis_min_length)
```